internal/buildscripts/compare-apidiff.sh

```diff
@@ -33,12 +38,23 @@ if [ -z $package ]; then
   usage
 fi
 
+changes_found() {
+  echo "Incompatible Changes Found."
+  echo "Check the logs in the GitHub Action log group: 'Compare-States'."
+  exit 1
+}
+
 set -e
 
-if [ -d $input_dir/$package ]; then
+if [ -e $input_dir/$package/apidiff.state ]; then
   changes=$(apidiff $input_dir/$package/apidiff.state $package)
   if [ ! -z "$changes" -a "$changes"!=" " ]; then
-    echo "Changes found in $package:"
-    echo "$changes"
+    SUB='Incompatible changes:'
+    if [ $check_only = true ] && [[ "$changes" =~ .*"$SUB".* ]]; then
+      changes_found
```

Comment:
Out of curiosity: why did you prefer to externalize this? It's being called only once, and the code seems quite simple, I would say that it's more readable to have it here, no?

Reply:
I initially externalized it, because I thought it made it more readable. However, you bring up a great point.